helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AbstractSchemaBuilder.java

```diff
@@ -250,6 +260,49 @@ public Schema build() {
         return result;
     }
 
+    private Schema fixInnerNamespace(Schema schema, String parentNamespace) {
+        Schema result = schema;
+
+        switch (schema.getType()) {
+            case FIXED:
+            case ENUM:
+            case RECORD:
+                if (schema.getNamespace() == null && parentNamespace != null) {
+                    result = _adapter.newSchemaBuilder(schema).setNamespace(parentNamespace).build();
+                }
+                break;
+            case ARRAY:
+                if (parentNamespace != null) {
+                    result = _adapter.newSchemaBuilder(null)
+                        .setType(Schema.Type.ARRAY)
+                        .setElementType(schema.getElementType())
+                        .setNamespace(parentNamespace)
+                        .build();
+                }
+                break;
+            case MAP:
+                if (parentNamespace != null) {
+                    result = _adapter.newSchemaBuilder(null)
+                        .setType(Schema.Type.MAP)
+                        .setValueType(schema.getValueType())
```

Comment:
you might need to recurse into the value type.

Reply:
**recurse into the value type** happens in the build() method.